diesel/src/query_source/joins.rs

```diff
@@ -292,6 +261,22 @@ impl<T, Join, On> AppearsInFromClause<T> for JoinOn<Join, On> where
     type Count = Join::Count;
 }
 
+#[allow(missing_debug_implementations, missing_copy_implementations)]
+#[doc(hidden)]
+/// A hack to allow bidirectional joins to be generated from `#[belongs_to]`
+///
+/// This type needs to exist because it is illegal in Rust today to write
```

Comment:
This should be possible in a post-chalk world. Would not be possible if `User` was a type parameter, but after chalk we should be able to normalize this to `users` or w/e.